### PR TITLE
Suppression list for the build using Valgrind on ARM

### PR DESCRIPTION
**Pull request by asenyaev**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [ ] There is reference to original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### Description
This PR solves issues with leaks and uninitialized values on ARM.

### CODE REVIEW

**Comment by asenyaev** [comment]
jenkins cn please retry a build

**Review by asmorkalov — CHANGES_REQUESTED** [review]
The PR has been discussed off-line.

File: platforms/scripts/valgrind_3rdparty_arm.supp [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+   Memcheck:Leak
+   ...
+   fun:_ZNK2cv23StaticBackendWithParams12createWriterERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEidRKNS_5Size_IiEERKNS_21VideoWriterParametersE
+}
+
+{
+   vld1q_dup_u32
+   Memcheck:Value8
+   ...
+   fun:vld1q_dup_u32
```
[the remark this answers is not in the document]
Comment (asenyaev) [reply platforms/scripts/valgrind_3rdparty_arm.supp]: Removed it and keep only a function `png_do_expand_palette_rgb8_neon`.

File: platforms/scripts/valgrind_3rdparty_arm.supp [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+   Memcheck:Value8
+   ...
+   fun:png_read_image
+}
+
+{
+   vld3_dup_u8
+   Memcheck:Leak
+   ...
+   fun:vld3_dup_u8
```
[the remark this answers is not in the document]
Comment (asenyaev) [reply platforms/scripts/valgrind_3rdparty_arm.supp]: Removed it and keep only a function `png_do_expand_palette_rgb8_neon`.

File: platforms/scripts/valgrind_3rdparty_arm.supp [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+   ...
+   fun:png_read_row
+}
+
+{
+   png_read_image
+   Memcheck:Value8
+   ...
+   fun:png_read_image
+}
```
[the remark this answers is not in the document]
Comment (asenyaev) [reply platforms/scripts/valgrind_3rdparty_arm.supp:104]: I found that these 3 suppressions are used in different places in `3rdparty/libpng`, and we need to keep all of them. But I can merge it to the one suppression and define all fun, like this:
```
{
   libpng
   Memcheck:Value8
   ...
   fun:png_do_read_transformations
   fun:png_read_image
   fun:png_read_row
}
```

**Comment by alalek** [comment]
@asmorkalov Ping

**Pull request closed without merging**